tests/MaintainersMetadataTest.php

```diff
@@ -0,0 +1,222 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\ManageApiTest;
+
+use Keboola\ManageApi\Client;
+use Keboola\ManageApi\ClientException;
+
+class MaintainersMetadataTest extends ClientTestCase
+{
+    public const TEST_METADATA = [
+        [
+            'key' => 'test_metadata_key1',
+            'value' => 'testval',
+        ],
+        [
+            'key' => 'test.metadata.key2',
+            'value' => 'testval2',
+        ],
+    ];
+
+    public const PROVIDER_USER = 'user';
+    public const PROVIDER_SYSTEM = 'system';
+
+    /**
+     * @var array
+     */
+    private $maintainer;
+
+    public function setUp()
+    {
+        /*
+         * $this->client = super-admin
+         * $this->normalUserClient = normal admin without any connections yet
+         */
+        parent::setUp();
+
+        $maintainerPrefix = sprintf('%s - MaintainerMetadata', self::TESTS_MAINTAINER_PREFIX);
+
+        // cleanup maintainers created by tests
+        foreach ($this->client->listMaintainers() as $maintainer) {
+            if (strpos($maintainer['name'], $maintainerPrefix) === 0) {
+                $this->client->deleteMaintainer($maintainer['id']);
```

Comment:
Toto tu asi nemusi byt v     parent::setUp(); sa mazu vsetky maintainers vytvoreny v testoch

Reply:
OK, jsem to z toho nejak nepochopil... zkusim dat pryc. Ale mam pocit, ze to maze pro jineho maintainera ($this->testMaintainerId, ale ja tu mam svyho.